examples/src/components/AttributeFilterExample.jsx

```diff
@@ -29,41 +34,87 @@ export class AttributeFilterExample extends Component {
         console.info("AttributeFilterExample onLoadingChanged", ...params);
     }
 
-    onApply(filter) {
+    onApply = filter => {
         // eslint-disable-next-line no-console
         console.log("AttributeFilterExample onApply", filter);
-        const isPositiveFilter = AFM.isPositiveAttributeFilter(filter);
+        this.setState({ filters: [], error: null });
+        if (filter.in) {
+            this.filterPositiveAttribute(filter);
+        } else {
+            this.filterNegativeAttribute(filter);
+        }
+    };
+
+    onApplyWithFilterDefinition = filter => {
+        // eslint-disable-next-line no-console
+        console.log("AttributeFilterExample onApplyWithFilterDefinition", filter);
+        const isPositiveFilter = VisualizationInput.isPositiveAttributeFilter(filter);
         const inType = isPositiveFilter ? "in" : "notIn";
         const filterItems = isPositiveFilter
             ? filter.positiveAttributeFilter[inType]
             : filter.negativeAttributeFilter[inType];
 
-        if (!filterItems.length) {
-            if (isPositiveFilter) {
-                this.setState({
-                    error: "The filter must have at least one item selected",
-                });
-            } else {
-                this.setState({ filters: [], error: null });
-            }
+        if (!filterItems.length && isPositiveFilter) {
+            this.setState({
+                error: "The filter must have at least one item selected",
+            });
         } else {
             this.setState({ filters: [filter], error: null });
         }
-    }
+    };
 
     onError(...params) {
         // eslint-disable-next-line no-console
         console.info("AttributeFilterExample onLoadingChanged", ...params);
     }
 
+    filterPositiveAttribute(filter) {
+        let filters;
+        if (filter.in.length !== 0) {
+            filters = [
+                {
+                    positiveAttributeFilter: {
+                        displayForm: {
+                            identifier: filter.id,
+                        },
+                        in: filter.in.map(element => `${locationResortUri}/elements?id=${element}`),
+                    },
+                },
+            ];
+        } else {
+            this.setState({
```

Comment:
what about doing `return` when this error state appears? (so that we don't call setState with filters undef)

Reply:
See it's consistent with state before merge of #1170, closing